awsrequests/__init__.py

```diff
@@ -122,15 +115,18 @@ def request(self, method, url,
                                hooks=hooks
                                )
         prepped = req.prepare()
-        aws_auth_headers = get_headers_for_request(url,
-                                                   self.region,
-                                                   'execute-api',
-                                                   self.access_key,
-                                                   self.secret_key,
-                                                   self.session_token,
-                                                   payload=prepped.body,
-                                                   method=prepped.method,
-                                                   t=time)
+        aws_auth_headers = get_headers_for_request(
+                url,
+                self.args['region'],
+                'execute-api',
+                self.temp_creds["Credentials"]["AccessKeyId"],
```

Comment:
You can't force using temp creds here.  This library used to just piggy-back off boto, so you could use real creds or meta-data provided creds or whatever.  This forces to uses STS, which is not right